caffe2/core/allocator.h

```diff
@@ -42,6 +43,8 @@ class CAFFE2_API MemoryAllocationReporter {
   std::mutex mutex_;
   std::unordered_map<void*, size_t> size_table_;
   size_t allocated_;
+  size_t allocations_;
+  size_t deallocations_;
```

Comment:
I didn't see you initialize these to zero anywhere in the patch; they need to be explicitly initialized (I recommend putting `= 0` here)